xml-format.md

```diff
@@ -0,0 +1,207 @@
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This specification promotes an attribute heavy XML document style, such a model
+presents advantages in terms of parsing/processing along with a more compact
+XML document representation.
+
+Attribute and Element names are deliberately terse for performance reasons.
+
+Preservation of attribute value type information is supported allowing
+custom extensions to be communicated without value type loss.
+
+The model is formalized by this [XML Schema][xml-spec].
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to XML schema types as follows :
+
+| CloudEvents   | XML | Type Tag |
+| ------------- | ---------------------------------------------------------------------- | -- |
+| Boolean       | [xs:boolean][xml-primitives] | B |
+| Integer       | [xs:int][xml-primitives] | I |
+| String        | [xs:string][xml-primitives] | S |
+| Binary        | [xs:base64Binary][xml-primitives] | BI |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| U |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | UR |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | T |
+
+Each concrete CloudEvent context attribute is represented using an XML element with the following
+shape:
+
+``` xml
+<Attr n="id" v="XXX-YYY-ZZZ-0000" t="S">
+```
+
+where :
+
+* _n_ is the _name_ of the context attribute
+* _v_ is the _value_ of the context attribute
+* _t_ is the _type_ of the attribute value acording to the _Tag_ defined in the table above.
+
+See section below on special handling for `specversion`
+
+## 2.1 Specification Version
+
+To better support streaming style processing the specification version
+(`specversion` context attribute) statement is communicated in an explicit element.
+
+``` xml
+<Ver>1.0</Ver>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent is contained is a `<Data></Data>` element and supports
+Binary, Text, and XML Data. Each represention is wrapped in a discriminating element:
+
+### Binary Data
+
+Carried in an element with an expected type of `xs:base64Binary`
+
+``` xml
+<Data>
+    <Bin>... base64 encoded string ...</Bin>
+</Data>
+```
+
+### Text Data
+
+Carried in an element with an expected type of `xs:string`
+
+``` xml
+<Data>
+    <Txt>Hello World</Txt>
+</Data>
+```
+
+### XML Data
+
+Carried in an element with an expected type of `xs:any`
+
+``` xml
+<Data>
+    <Xml>
+        <SomeElement>
+            ....
+        </SomeElement>
+    </Xml>
+</Data>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element, the root xml
+document element is `<CloudEvent ver="n.n">` with the `ver` attribute
+used to define the specification version being followed.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of context attributes.
+* An optional data element.
```

Comment:
Any restriction on whether the data element is permitted/required based on the presence of a `<Attr n="datacontenttype" ...>` attribute? (That's something I've found myself wondering in a few places.)

Reply:
I don't think the other formats make this type of connection especially since that is an optional context attribute, I think it's more of a usage expectation - I'll try and come up with some language.